csharp/TheVehicleAbstractClass/TheVehicleAbstractClass/Vehicles/Vehicle.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TheVehicleAbstractClass.Vehicles
+{
+    public abstract class Vehicle
+    {
+        private static int _totalRegistrations = 0;
+        private int _registrationNumber;
+        private int _year;
+        private double _price;
+
+        public int RegistrationNumber
+        {
+            get { return _registrationNumber; }
+        }
+
+        public int Year
+        {
+            get { return _year; }
+            set { _year = value; }
+        }
+
+        public double Price
+        {
+            get { return _price; }
+            set { _price = value; }
+        }
+
+        public static int TotalRegistrations
+        {
+            get { return _totalRegistrations; }
+        }
+
+        public Vehicle(int year, double price)
+        {
+            _registrationNumber = ++_totalRegistrations;
```

Comment:
this is totally fine, just be aware that in a multi-thread environment you would need to manage this differently.